crates/rattler_conda_types/src/conda_lock/builder.rs

```diff
@@ -0,0 +1,209 @@
+                .collect(),
+            version: super::default_version(),
+        };
+        Ok(lock)
+    }
+}
+
+/// Shorthand for creating packages per platform
+struct LockedPackages {
+    pub locked_packages: Vec<LockedPackage>,
+    pub platform: Platform,
+}
+
+impl LockedPackages {
+    pub fn new(platform: Platform) -> Self {
+        Self {
+            locked_packages: Vec::new(),
+            platform,
+        }
+    }
+
+    /// Add a locked package
+    pub fn add_locked_package(mut self, locked_package: LockedPackage) -> Self {
+        self.locked_packages.push(locked_package);
+        self
+    }
+
+    /// Transform into list of [`LockedDependency`] objects
+    pub fn build(self) -> Vec<LockedDependency> {
+        self.locked_packages
+            .into_iter()
+            .map(|locked_package| {
+                LockedDependency {
+                    name: locked_package.name,
+                    version: locked_package.version,
+                    /// Use conda as default manager for now
+                    manager: Manager::Conda,
+                    platform: self.platform,
+                    dependencies: locked_package.dependency_list,
+                    url: locked_package.url,
+                    hash: locked_package.package_hashes,
+                    optional: locked_package.optional.unwrap_or(false),
+                    category: super::default_category(),
+                    source: None,
+                    build: Some(locked_package.build_string),
+                }
+            })
+            .collect()
+    }
+}
+
+/// Short-hand for creating a LockedPackage that transforms into a [`LockedDependency`]
+struct LockedPackage {
+    pub name: String,
+    pub version: String,
+    pub build_string: String,
+    pub url: Url,
+    pub package_hashes: PackageHashes,
+    pub dependency_list: HashMap<String, VersionConstraint>,
+    pub optional: Option<bool>,
+}
+
+impl LockedPackage {
+    /// Set if the package should be optional
+    pub fn set_optional(mut self, optional: bool) -> Self {
+        self.optional = Some(optional);
+        self
+    }
+
+    /// Add a single dependency
+    pub fn add_dependency<S: AsRef<str>>(
+        mut self,
+        key: S,
+        version_constraint: VersionConstraint,
+    ) -> Self {
+        self.dependency_list
+            .insert(key.as_ref().to_string(), version_constraint);
+        self
+    }
+
+    /// Add multiple dependencies
+    pub fn add_dependencies(
+        mut self,
+        value: impl IntoIterator<Item = (String, VersionConstraint)>,
+    ) -> Self {
+        self.dependency_list.extend(value);
+        self
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::conda_lock::builder::{LockFileBuilder, LockedPackage, LockedPackages};
+    use crate::conda_lock::{Channel, CondaLock, PackageHashes};
+    use crate::{ChannelConfig, MatchSpec, Platform};
+
+    #[test]
+    fn create_lock_file() {
```

Comment:
This function doesnt actually test anything :)

Reply:
Yeah I kinda wanted it to function as an identity function